includes/Framework/PaymentGateway/Integrations/Payment_Gateway_Integration_Subscriptions.php

```diff
@@ -220,24 +227,34 @@ public function save_payment_meta( $order ) {
 	 * @param \WC_Order $order original order containing the subscription
 	 */
 	public function process_renewal_payment( $amount_to_charge, $order ) {
+		Performance_Logger::start( 'process_renewal_payment', $this->get_gateway()->get_plugin() );
 
-		// set payment total so it can override the default in get_order()
-		$this->renewal_payment_total = Square_Helper::number_format( $amount_to_charge );
+		try {
 
-		$token = $this->get_gateway()->get_order_meta( Order_Compatibility::get_prop( $order, 'id' ), 'payment_token' );
+			// set payment total so it can override the default in get_order()
+			$this->renewal_payment_total = Square_Helper::number_format( $amount_to_charge );
 
-		// payment token must be present and valid
-		if ( empty( $token ) || ! $this->get_gateway()->get_payment_tokens_handler()->user_has_token( $order->get_user_id(), $token ) ) {
+			$token = $this->get_gateway()->get_order_meta( Order_Compatibility::get_prop( $order, 'id' ), 'payment_token' );
 
-			$this->get_gateway()->mark_order_as_failed( $order, esc_html__( 'Subscription Renewal: payment token is missing/invalid.', 'woocommerce-square' ) );
+			// payment token must be present and valid
+			if ( empty( $token ) || ! $this->get_gateway()->get_payment_tokens_handler()->user_has_token( $order->get_user_id(), $token ) ) {
+				Performance_Logger::end( 'process_renewal_payment', $this->get_gateway()->get_plugin(), true );
 
-			return;
-		}
+				$this->get_gateway()->mark_order_as_failed( $order, esc_html__( 'Subscription Renewal: payment token is missing/invalid.', 'woocommerce-square' ) );
+
+				return;
+			}
 
-		// add subscriptions data to the order object prior to processing the payment
-		add_filter( 'wc_payment_gateway_' . $this->get_gateway()->get_id() . '_get_order', array( $this, 'get_order' ) );
+			// add subscriptions data to the order object prior to processing the payment
+			add_filter( 'wc_payment_gateway_' . $this->get_gateway()->get_id() . '_get_order', array( $this, 'get_order' ) );
 
-		$this->get_gateway()->process_payment( Order_Compatibility::get_prop( $order, 'id' ) );
+			$this->get_gateway()->process_payment( Order_Compatibility::get_prop( $order, 'id' ) );
+
+			Performance_Logger::end( 'process_renewal_payment', $this->get_gateway()->get_plugin() );
+		} catch ( \Exception $e ) {
+			Performance_Logger::end( 'process_renewal_payment', $this->get_gateway()->get_plugin(), true );
+			throw $e;
```

Comment:
And here

Reply:
Return failed order in https://github.com/woocommerce/woocommerce-square/pull/279/commits/b957e87a7acbab22a85917dc33ed2d5197e95c62